src/main.rs

```diff
@@ -82,6 +83,54 @@ pub fn insertion(tree: &mut Smt, size: u64) -> Result<(), MerkleError> {
     Ok(())
 }
 
+pub fn batched_insertion(tree: &mut Smt, size: u64) -> Result<(), MerkleError> {
```

Comment:
should we use `Criterion` for benchmarking?

Reply:
I think it is OK to keep it `main.rs` for now. We can refactor how we do benchmarking separately, if we decide to do it.